R/AddCardModule.R

```diff
@@ -89,7 +89,13 @@ add_card_button_srv <- function(id, reporter, card_fun) {
         shiny::modalDialog(
           easyClose = TRUE,
           shiny::tags$h3("Add a Card to the Report"),
-          shiny::tags$hr(),
+          shiny::textInput(
+            ns("label"),
+            "Card Name",
+            value = "",
```

Comment:
It feels like the modules should be able to give a default to here?

Reply:
this requires passing `label` as an argument to `add_card_button_srv` and `simple_reporter_srv`.

Reply:
Yup that's definitely a better way of adding it - though we'd still need to pass the label into the module args - might be best to not do that right now though and wait until teal is reconfigured to make that not needed...

Reply:
> Another way can be to not have any default for label then to have empty name if not specified.when card is added.

Or to have the name the module developer has set for it (rather than the app developer).

Reply:
so the latest update is:
1) the module developer can set a default card label in the card function
2) the module user can change the card label with the UI input
3) No support for passing the module name by default to the reporter cards till the teal refactor is accomplished as it requires passing the label argument to the server function of all modules.